overlord/snapstate/handlers.go

```diff
@@ -1342,6 +1342,27 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 		snapst.Sequence[len(snapst.Sequence)-1] = cand
 	}
 
+	if snapsup.Revert {
+		// for undo; save it early so we don't have to make a deep copy
+		// before modifying it.
+		t.Set("old-revert-status", snapst.RevertStatus)
```

Comment:
it would be slightly better I think if this was reorganized so that the `t.Set("old-revert-status", oldRevertStatus)` would be together with the other `old-xxx` saving Set calls.

Reply:
Done.